cardano-db-sync/src/Cardano/DbSync/Metrics.hs

```diff
@@ -4,25 +4,52 @@
 module Cardano.DbSync.Metrics
   ( Metrics (..)
   , makeMetrics
+  , withMetricsLayer
   , withMetricsServer
   ) where
 
 import           Cardano.Prelude
 
+import           Cardano.Sync.Types (MetricsLayer (..))
+
 import           System.Metrics.Prometheus.Concurrent.RegistryT (RegistryT (..), registerGauge,
                    runRegistryT, unRegistryT)
 import           System.Metrics.Prometheus.Http.Scrape (serveMetricsT)
 import           System.Metrics.Prometheus.Metric.Gauge (Gauge)
+import qualified System.Metrics.Prometheus.Metric.Gauge as Gauge
 
-
+-- |The metrics we use for Prometheus.
 data Metrics = Metrics
```

Comment:
Why do we have both a `Metrics` struct and a `MetricsLayer` ?

And why is it a layer? To me "layer" implies that if we have say 3 layers, `A`, `B` and `C`, that `A` calls into `B` and then `B` calls into `C`. That is not the case here. Its not a layer, its a Metrics component.

Reply:
The term `Layer` has been used throughout the codebase in all these instances. Please, let's use the name `Layer` that is a standard term for these things. A `Setter` is not. 
Examples:
- https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Pool/DB.hs#L15
- https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/Network.hs#L15
- https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/Transaction.hs#L21